darwin/future/meta/objects/stage.py

```diff
@@ -75,6 +83,30 @@ def move_attached_files_to_stage(self, new_stage_id: UUID) -> Stage:
             self.meta_params["dataset_id"],
         )
         ids = [x.id for x in self.item_ids.collect_all()]
+
+        def all_items_complete(item_ids: list[UUID]) -> bool:
+            for item_id in item_ids:
```

Comment:
I'd probably use `all` with a list comp here, as it's more performant - I don't know if we really expect this to be enough of a factor to matter, but given that it's an easy change, I'd suggest it.

```python
def all_items_complete(item_ids: list[UUID]) -> bool:
    return all(
        get_item(
            self.client,
            slug,
            item_id
        ).processing_status == "complete"
        for item_id in item_ids
    )
```

☝🏻 Check the above, it's untested, but it's a decent gist of the idea.

Reply:
Small thing to consider here as well is that each time this gets called, it's making len(item_ids) requests, regardless of how many are still processing. 

I would propose a check that pops off from the list ones that are completed so they're not continually checked.

Reply:
Yeah, decent idea, could make the difference at scale.